### ISSUE TITLE
Add contribution guidelines and issue/PR templates

### ISSUE THREAD

**Issue by franz-josef-kaiser** [wecodemore/wpstarter#25]
- [x] Contribution guidelines
- [x] Pull request template
- [x] Issue template
- [x] Agree on commit rules
- [x] Agree on contributing guidelines

**Comment by franz-josef-kaiser** [comment]
@Giuseppe-Mazzapica I worked that one out for `grunt-githooks` actually and just adapted it. Not sure if the _Angular_ commit style here makes much sense. At least not without any explanation of what the `<topic>` actually can be.

**Comment by GaryJones** [comment]
The `Contributions` section in the Readme should link to the `.github/CONTRIBUTING.md` file.

**Comment by franz-josef-kaiser** [comment]
[Link to Contributing](https://github.com/wecodemore/wpstarter/blob/master/README.md#contributing).

**Comment by Giuseppe-Mazzapica** [comment]
@franz-josef-kaiser Honestly, I really dislike Angular commit message format.

`scope` and `subject` are arbitrary most of the times and they remove space to textual content of the commit messages.

First line of commit message has to be less than 50 characters. I prefer to use all of them to make the message as more human readable and self explanatory as possible in plain english. And `<type>(<scope>) <subject>` sometimes takes half of the space and without helping. Unless you know very well the project, but commit messages have to be readable for everyone.

Moreover, a good commit message explain _why_ something has been done, but Angular commit message format is strongly focused in _what_ has be done.

So if it would be my decision, I would just remove that part from guidelines, but I am open to change my mind if the majority of people who contribute here like them.

**Plese use reaction to this comment**:
- :+1: means "NO Angular commit message format"
- :-1: means "Let's use Angular commit message format"

**Comment by gmazzap** [comment]
Ok there's an agreement here. I will add the files in master and other branch.

**Comment by gmazzap** [comment]
The repo has all the template in place. Thanks anyone.
